apis/core/bootstrap/shapes/csv-source.ts

```diff
@@ -26,9 +34,91 @@ ${shapeCreateId} {
       ${sh.minCount} 1 ;
       ${sh.maxCount} 1 ;
     ] ;
+    ${sh.xone} (
+      [
+        ${sh.closed} true ;
+        ${sh.ignoredProperties} (
+          ${cc.sourceKind}
+          ${schema.name}
+          ${schema.contentUrl}
+          ${rdf.type}
+        ) ;
+        ${sh.property} [
+          ${sh.path} ${cc.sourceKind} ;
+          ${sh.hasValue} ${cc.MediaLocal} ;
+          ${dash.hidden} true ;
+        ] ;
+        ${sh.property} [
+          ${sh.name} "S3 key" ;
+          ${sh.path} ${schema.identifier} ;
+          ${sh.datatype} ${xsd.string} ;
+          ${sh.minCount} 1 ;
+          ${sh.maxCount} 1 ;
+        ] ;
+      ]
+      [
+        ${sh.closed} true ;
+        ${sh.ignoredProperties} (
+          ${cc.sourceKind}
+          ${schema.name}
+          ${schema.contentUrl}
+          ${rdf.type}
+        ) ;
+        ${sh.property} [
+          ${sh.path} ${cc.sourceKind} ;
+          ${sh.hasValue} ${cc.MediaURL} ;
+          ${dash.hidden} true ;
+        ] ;
+      ]
+      [
+        ${sh.closed} true ;
```

Comment:
I already included the shape for external S3 bucket even though it's not supported yet.